website/docs/basics/blocks-component-in-block.md

```diff
@@ -342,3 +342,303 @@ echo wp_kses_post(
+
+```json
+{
+  "attributes": {
+    "typographySize": {
+      "type": "string",
+      "default": "16-text-roman"
+    },
+    "typographyColor": {
+      "type": "string",
+      "default": "black",
+      "variable": true,
+      "color": true
+    },
+  },
+  "options": {
+    "typographySize": [
+      {
+        "label": "180 Display",
+        "value": "180-default"
+      },
+      {
+        "label": "120 Display",
+        "value": "120-default"
+      },
+      {
+        "label": "80 Display",
+        "value": "80-default"
+      },
+      {
+        "label": "52 Display",
+        "value": "52-default"
+      },
+      {
+        "label": "36 Text",
+        "value": "36-text"
+      }
+    ],
+    "typographyColor": [
+      "black",
+      "white",
+      "grey100",
+      "grey200"
+    ]
+  }
+}
+```
+
+2. Each option in your component must have `getOptions` helper used in the prop that is used to provide options. Details about the helper can be found [here](helpers-javascript#getoptions).
+
+**SelectControl Example:**
+```js
+<SelectControl
+  label={
+    <>
+      <Icon icon={icons.textSize} />
+      {__('Text size', 'eightshift-boilerplate')}
+    </>
+  }
+  value={typographySize}
+  options={getOptions(manifest, componentName, 'size', options)}
+  onChange={(value) => setAttributes({ [`${componentName}Size`]: value })}
+/>
+```
+
+**ColorPaletteCustom Example:**
+```js
+<ColorPaletteCustom
+  label={
+    <>
+      <Icon icon={icons.color} />
+      {__('Color', 'eightshift-boilerplate')}
+    </>
+  }
+  colors={getOptionColors(getOptions(manifest, componentName, 'color', options))}
+  value={typographyColor}
+  onChange={(value) => setAttributes({ [`${componentName}Color`]: value })}
+/>
+```
+
+3. In the component/blocks that you are going to override options, you must provide an options prop that is going to override the default one.
+
+**heading-options.js**
+```js
+<HeadingOptions
+  options={options}
+  // ...
+/>
+```
+
+4. In the component/blocks that you are going to override options, you must provide the override in the `manifest.json` by following the same naming as in the component.
+
+```json
+{
+  "options": {
+    "headingSize": [
+      "80-default",
+      "52-default"
+    ],
+    "headingColor": [
+      "black",
+      "white"
+    ]
+  }
+}
+```
+
+An this is it you are now able to override options from the parent block/component.
+
+> Keep in mind that you can only override SelectControl, ColorPaletteCustom, and AlignmentToolbar.
+
+### I want to only pass attributes to the component that I'm going to use
+
+At one point in time, we agreed on naming standards for all component attributes and that way we are sure that you wouldn't get any collisions when using multiple components. So we said it is ok to spread all attributes to the component and let the component handle what it needs. Well, that approach is ok but it can bite you in the a.. at the point that you least expect.
```

Comment:
```suggestion
At one point in time, we agreed on naming standards for all component attributes. That way we made sure that you wouldn't get any collisions when using multiple components. So we said it is ok to spread all attributes to the component and let the component handle what it needs. Well, that approach is ok but it can bite you in the a.. at the point that you least expect.
```